tst/testinstall/methwhy.tst

```diff
@@ -91,124 +136,207 @@ function ( x, y... )
+#I   at stream:2
 function ( x, y... )
     return 1;
 end
 
 # check the method for two lists
 gap> ApplicableMethod(foobar, [[1], []], 5);;
 #I  Searching Method for foobar with 2 arguments:
-#I  Total: 2 entries
-#I  Method 1: ``foobar: for two lists'' at stream:1, value: 17
+#I  Total: 2 entries, of which 1 is applicable:
+#I  Method 1, applicable method number 1, value: 17
+#I  ``foobar: for two lists''
+#I   at stream:1
 
 # no method found
-gap> ApplicableMethod(foobar, [1, []], 5);
+gap> ApplicableMethod(foobar, [1, []], 1);   
 #I  Searching Method for foobar with 2 arguments:
-#I  Total: 2 entries
-#I  Method 1: ``foobar: for two lists'' at stream:1, value: 17
+#I  Total: 2 entries, of which 0 are applicable:
+#I  there are no applicable methods with these parameters
+fail
+gap> ApplicableMethod(foobar, [1, []], 2);
+#I  Searching Method for foobar with 2 arguments:
+#I  Total: 2 entries, of which 0 are applicable:
+#I  there are no applicable methods with these parameters
+#I  Method 1, value: 17
+#I  ``foobar: for two lists''
+#I   at stream:1
+#I  Method 2, value: 0
+#I  ``foobar: for 2 integers''
+#I   at stream:2
+fail
+gap> ApplicableMethod(foobar, [1, []], 4);
+#I  Searching Method for foobar with 2 arguments:
+#I  Total: 2 entries, of which 0 are applicable:
+#I  there are no applicable methods with these parameters
+#I  Method 1, value: 17
+#I  ``foobar: for two lists''
+#I   at stream:1
 #I   - 1st argument needs [ "IsList" ]
-#I  Method 2: ``foobar: for 2 integers'' at stream:2, value: 0
+#I  Method 2, value: 0
+#I  ``foobar: for 2 integers''
+#I   at stream:2
 #I   - 2nd argument needs [ "IsInt" ]
 fail
 
 #
 # check 0-6 arguments
 #
-gap> Display(ApplicableMethod(foobar, [], 5));
+gap> Display(ApplicableMethod(foobar, [], 4));
 #I  Searching Method for foobar with 0 arguments:
-#I  Total: 1 entries
-#I  Method 1: ``foobar: for no argument'' at stream:1, value: 0
+#I  Total: 1 entries, of which 1 is applicable:
+#I  Method 1, applicable method number 1, value: 0
+#I  ``foobar: for no argument''
+#I   at stream:1
 function (  )
     return 1;
 end
-gap> Display(ApplicableMethod(foobar, [1], 5));
+gap> Display(ApplicableMethod(foobar, [1], 4, 1));
 #I  Searching Method for foobar with 1 arguments:
-#I  Total: 4 entries
-#I  Method 1: ``foobar: for a string'' at stream:1, value: 2*SUM_FLAGS+1
+#I  Total: 4 entries, of which 2 are applicable:
+#I  Method 1, value: 2*SUM_FLAGS+1
+#I  ``foobar: for a string''
+#I   at stream:1
 #I   - 1st argument needs [ "IsString" ]
-#I  Method 2: ``foobar: for a list'' at stream:1, value: 42
+#I  Method 2, value: 42
+#I  ``foobar: for a list''
+#I   at stream:1
 #I   - 1st argument needs [ "IsList" ]
-#I  Method 3: ``foobar: for an integer'' at stream:1, value: 1
+#I  Method 3, applicable method number 1, value: 1
+#I  ``foobar: for an integer''
+#I   at stream:1
 function ( x )
     return 1;
 end
-gap> Display(ApplicableMethod(foobar, [1,2], 5));
+gap> ApplicableMethod(foobar, [1], 4, 2);
```

Comment:
Dropping the `Display` here and elsewhere in the tests has two major disadvantages:
* having the function printed served as another verification that we *really* show the right function; this is now lost;
* it results in a larger diff, which makes reviewing more painful; and I think the huge and hard to read diff is a major reason your PR has not received more attention (it certainly is the reason why I don't really feel like reviewing it).